docs/Developers/Design/CRD_design.md

```diff
@@ -0,0 +1,69 @@
+This document describes the set of [Custom Resource Definitions
+(CRDs)](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)
+that will be used to deploy mixins for a storage system. The actual implementation
+of the resources described here will be phased in during development. The
+purpose of this document is to provide the overall structure, ensuring the end
+result provides necessary configurability in a user-friendly manner.
+
+# Overview
+
+A single Monstorak operator can take care of deployment of mixins for one or
+more storage systems. Each storage cluster should already be deployed within
+the kubernetes cluster, or the storage cluster could be running on hosts outside
+of kubernetes cluster. The capabilities of the operator remains same in both
+the modes of the deployment of storage cluster and the same set of CRDs should
+be used for both the cases.
+
+A given deployment configuration for storage systems contains a Custom Resource
+which contains configurations options like whether alerts are enabled, if
+dashboards are enabled, name of the storage system and version details of storage
+system.
+
+![Hierarchy of monstorak custom resources](crd_hierarchy.dot.png)
+
+# Custom Resource Definitions
+
+This section describes the fields in each of the custom resources.
+All CRs live within the `mixins.monstorak.org` group and have version
+`v1alpha1`.
+
+## **StorageAlert**
+
+### *Singular: StorageAlert, Plural:StorageAlerts*
+
+The StorageAlert custom resource defines a storage system for which mixins could
+be deployed using the operator. This CR captures the names of the storage system,
+its supported version details, namespace in which alerts should be deployed.
+
+```YAML
+apiVersion: mixins.monstorak.org/v1alpha1
+kind: StorageAlert
```

Comment:
```
kind: StorageAlert
spec:
  - storage: "ceph"
    - version: "mimic"
      alert: false
    - version: "nautilus"
      alert: true
      prometheus:
      - namespace: "monitoring"
      - label: "prometheus:k8s"
```

I thought this was the proposed structure. @aruniiird Looks like even you are using older CRD structure.

Reply:
@aruniiird if you confirm, I will change to this.
@umangachapagain I checked from `_types.go` and found this so added